src/relayserver/RelayServer.js

```diff
@@ -270,39 +311,70 @@ class RelayServer extends EventEmitter {
       })
   }
 
+  fatal (message) {
+    console.error('FATAL: ' + message)
+    process.exit(1)
+  }
+
+  async _init () {
+    const relayHubAddress = this.relayHubContract.options.address
+    console.log('Server address', this.managerAddress)
+    const code = await this.web3.eth.getCode(relayHubAddress)
+    if (code.length < 10) {
+      this.fatal(`No RelayHub deployed at address ${relayHubAddress}.`)
+    }
+    const version = await this.relayHubContract.methods.getVersion().call().catch(e => 'no getVersion() method')
+    if (version !== '1.0.0') {
+      this.fatal(`Not a valid RelayHub at ${relayHubAddress}: version: ${version}`)
+    }
+    const stakeManagerAddress = await this.relayHubContract.methods.getStakeManager().call()
+    this.stakeManagerContract = new this.web3.eth.Contract(StakeManagerABI, stakeManagerAddress)
+    const stakeManagerTopics = [Object.keys(this.stakeManagerContract.events).filter(x => (x.includes('0x')))]
+    this.topics = stakeManagerTopics.concat([['0x' + '0'.repeat(24) + this.managerAddress.slice(2)]])
+
+    this.chainId = await this.web3.eth.getChainId()
+    this.networkId = await this.web3.eth.net.getId()
+    if (this.devMode && (this.chainId < 1000 || this.networkId < 1000)) {
+      console.log('Don\'t use real network\'s chainId & networkId while in devMode.')
+      process.exit(-1)
+    }
+    this.initialized = true
+  }
+
+  async replenishWorker (workerIndex) {
+    const workerAddress = this.getAddress(workerIndex)
+    const workerBalance = toBN(await this.web3.eth.getBalance(workerAddress))
+    if (workerBalance.lt(toBN(this.workerMinBalance))) {
+      const refill = toBN(this.workerTargetBalance).sub(workerBalance)
+      console.log(`== replenishWorker(${workerIndex}): mgr balance=${this.balance.toString() / 1e18} worker balance=${workerBalance.toString() / 1e18} refill=${refill.toString() / 1e18}`)
+      if (refill.lt(this.balance.sub(toBN(minimumRelayBalance)))) {
+        await this._sendTransaction({
+          signerIndex: 0,
+          destination: workerAddress,
+          value: refill,
+          gasLimit: 300000
+          // gasPrice:1
+        })
+        this.balance = this.balance.sub(refill)
```

Comment:
You should use refreshBalance() since this doesn't take into account the tx cost other than value sent.

Reply:
fixed